optimade/server/entry_collections.py

```diff
@@ -120,13 +120,29 @@ def find(
         else:
             fields = all_fields.copy()
         results = []
+
+        # build up set of included data with unique IDs
+        included = []
+        id_set = set()
         for doc in self.collection.find(**criteria):
             results.append(self.resource_cls(**self.resource_mapper.map_back(doc)))
+            for reference in (
```

Comment:
It seems to me that this can be done more "elegantly" utilizing the set methods, although that may not be the case.
Even so, I may try to do something here and add a commit, if it's possible.

Reply:
I've tidied it a little in my last commit